great_tables/_boxhead.py

```diff
@@ -12,7 +12,10 @@
 
 
 def cols_label(
-    self: GTSelf, cases: dict[str, str | BaseText] | None = None, **kwargs: str | BaseText
+    self: GTSelf,
+    cases: dict[str, str | BaseText] | None = None,
+    fn: Callable[[str | BaseText], str | BaseText] | None = None,
```

Comment:
Thanks for taking this and running with it. I'm so sorry to send you down a rough path, but maybe we take `fn` out of here for now?

We're realizing that the combination of `fn=` and the ability to wrap inputs with BaseText containers is a bit tricky. Since we'd have to explain and figure out how to apply `fn` to things like `md("**some text**")`.

This seems resolvable, but I would punt on for the sake of getting `cols_label_with()` in. There's some cleanup across Great Tables we can likely do with BaseText and UnitStr/definition list, which I'm guessing will make situations like this easier to handle.